### ISSUE TITLE
typescript-estree@3.0.0: Supported Typescript Version

### ISSUE THREAD

**Issue by frederikdekegel** [typescript-eslint/typescript-eslint#2096]
**Repro**
running eslint against an Angular/Typescript project that uses typescript 3.8.3

**Expected Result**
no warning about the typescript version as it was the case with @typescript-eslint @ 2.34.0

**Actual Result**
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <3.8.0

YOUR TYPESCRIPT VERSION: 3.8.3

Please only submit bug reports when using the officially supported version.

**Additional Info**
The info about the supported typescript version on https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/typescript-estree suggests the supported typescript version is >=3.3.1 <3.10.0 (as can be read on  https://github.com/typescript-eslint/typescript-eslint/blob/master/README.md#supported-typescript-version).

However, I could understand that typescript >=3.8 is not supported for typescript-estree when I read https://github.com/typescript-eslint/typescript-eslint/issues/1436

So basically, I wondering now if the warning is incorrect or the doc is incorrect. Also, if necessary,  how can I disable this warning?

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `3.0.0` |
| `TypeScript`                           | `3.8.3` |
| `node`                                 | `12.16.3` |
| `npm`                                  | `6.14.5` |

**Comment by SimenB** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/2057
